src/scripts/generic_coverage.py

```diff
@@ -0,0 +1,200 @@
+    return mas_dict
+
+
+def clean_up_scope_terms(_term_dict: Dict[str, str], mas_dict: Dict[str, str], _scope: str) -> Dict[str, str]:
+    # Remove all superclasses of terms on the scope list (via subClassOf/ and part_of)
+    sparql.setQuery(get_superclass_value_query(list(_term_dict.values()), _scope))
+    super_ret = sparql.queryAndConvert()
+    for super_class in super_ret["results"]["bindings"]:
+        mas_dict.pop(super_class['super']['value'], None)
+    return mas_dict
+
+
+def get_term_leaves(term_list: List[str], _scope: str) -> Dict[str, Dict[str, str]]:
+    # Get terms under, connected with 'subClassOf' relation, given the term list via template file from Ubergraph
+    _term_leaves_dict = {}
+    sparql.setQuery(get_tissue_list_query(term_list, _scope))
+    ret = sparql.queryAndConvert()
+    for row in ret["results"]["bindings"]:
+        if row['organ_label']['value'] not in _term_leaves_dict.keys():
+            _term_leaves_dict.update(
+                {row['organ_label']['value']: {row['tissue']['value']: row['tissue_label']['value']}})
+        else:
+            _term_leaves_dict[row['organ_label']['value']].update(
+                {row['tissue']['value']: row['tissue_label']['value']})
+    return _term_leaves_dict
+
+
+def get_invalid_subclass_list(_term_dict: Dict[str, Dict[str, str]]) -> List[str]:
+    merged_subset = {k: v for key, value in _term_dict.items() for k, v in value.items() if key != value.get(k)}
+    invalid_subclass_list = []
+    for term, term_subset in _term_dict.items():
+        if term in merged_subset.values():
+            invalid_subclass_list.append(term)
+    return invalid_subclass_list
+
+
+def get_scope_query(ontology: str, scope_term: str) -> str:
+    """Returns the SPARQL query to retrieve multicellular anatomical structure terms with IRIs and labels
+
+    Args:
+        ontology (str): Ontology name, can be either "uberon" or "all"
+        scope_term (str): Scope of the slim
+
+    Returns:
+        output (str): Multicellular anatomical structure SPARQL query
+    """
+    # ?mas rdfs:isDefinedBy 'obo:uberon.owl'
+    filter_str = "FILTER(strstarts(str(?mas),'http://purl.obolibrary.org/obo/UBERON_'))" if ontology == 'uberon' else ""
+    _scope = scope_term if ":" in scope_term else f"<{scope_term}>"
+    return f"""
+        PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+        PREFIX owl: <http://www.w3.org/2002/07/owl#>
+        PREFIX UBERON: <http://purl.obolibrary.org/obo/UBERON_>
+        PREFIX CL: <http://purl.obolibrary.org/obo/CL_>
+        SELECT ?mas ?label
+        WHERE
+        {{
+          ?mas <http://www.w3.org/2000/01/rdf-schema#subClassOf>|<http://purl.obolibrary.org/obo/BFO_0000050> {_scope} .
+          ?mas rdfs:isDefinedBy <http://purl.obolibrary.org/obo/cl.owl> .
+          ?mas rdfs:label ?label. 
+          ?s <http://www.w3.org/2000/01/rdf-schema#subClassOf>|<http://purl.obolibrary.org/obo/BFO_0000050> ?mas.
+          FILTER(?s != ?mas)
```

Comment:
Why do you need to include this?

Reply:
I think we needed it back when we were calculating the organ term coverage in CAP, but I cannot justify the reason at the moment. I've just removed it for now